color_helper.sublime-settings

```diff
@@ -394,7 +394,7 @@
             "base_scopes": ["text.ass"],
             "scanning": ["constant.other.color"],
             "color_class": "ass_abgr",
-            "color_trigger": "(?i)&h"
+            "color_trigger": "(?i)&h|\\\\[1-4]?c"
```

Comment:
Oh, I totally forgot the output part.

Reply:
About the trailing `&`, it's not necessary and I prefer to not add them actually... In Aegisub, it's not added in the `[Style]` part but Aegisub doesn't seem to have a button to add \c tag (can be typed manually though).

Reply:
So I was thinking, with a trigger like this: `(?i)(?:&H|(?<=\\\\c)[a-f0-9]|(?<=\\\\[1-4]c)[a-f0-9])`, you could target the matching at either `&H` or the hex character right after `\<num>c`.

Then with a pattern like`(?:&H)?([0-9a-fA-f]{8}|[0-9a-fA-f]{6})(&|\b)`, you would parse the color portion (including trailing `&` if necessary. This would allow you to swap a color without destroying context.

![ass](https://user-images.githubusercontent.com/1055125/132427113-8536653f-ee71-4245-b12c-bb4b29a03a09.gif)

Reply:
> About the trailing `&`, it's not necessary and I prefer to not add them actually... In Aegisub, it's not added in the `[Style]` part but Aegisub doesn't seem to have a button to add \c tag (can be typed manually though).

You don't have to output them, but do you not want to parse them in to normalize them away?

Reply:
I guess if the intent is to eat the `\c` portion and throw it away (like I do with the trailing `&` above), then including `\c` as part of the actual color parsing is fine. I'm just making sure it is behaving as you intend. Ultimately, I'll let you make the call.

Reply:
Fixed in a89ea46.


https://user-images.githubusercontent.com/6594915/132429568-53a69125-3baa-49d6-baa2-069624a3a502.mp4

Except the trailing `&` part.

Reply:
Sounds like I got you thinking about it and working through it. Let me know when you feel like you're happy with everything.